Console/Command/Convert.php

```diff
@@ -223,42 +234,30 @@ private function getType(string $patch): ?string
         return null;
     }
 
-    /**
-     * @param string $patch
-     * @return array
-     */
-    private function getAffectedComponents(string $patch): array
-    {
-        preg_match_all('/(magento\/[a-z\-]*)/', $patch, $components);
-        if (isset($components[0])) {
-            return $components[0];
-        }
-        return [];
-    }
-
     /**
      * @param string|null $patchId
      * @return string[]
      */
-    private function getFileAndDescriptionFromPatchesJson(?string $patchId): array
+    private function getInfoFromPatchesJson(?string $patchId): array
     {
         if (isset($this->patchesJsonContent[$patchId])) {
-            $packageKeys = array_keys($this->patchesJsonContent[$patchId]);
-            $package = $packageKeys[0];
+            $packageKeys = $this->filterPackageKeys(array_keys($this->patchesJsonContent[$patchId]));
+            $package = reset($packageKeys);
             $description = array_key_first($this->patchesJsonContent[$patchId][$package]);
-            $versionConstraint = array_key_first($this->patchesJsonContent[$patchId][$package][$description]);
-            $file = $this->patchesJsonContent[$patchId][$package][$description][$versionConstraint]['file'];
+            $constraint = $this->getValidConstraint($patchId, $package, $description);
+            $file = $this->patchesJsonContent[$patchId][$package][$description][$constraint]['file'];
 
             // See if the file found matches for the current Magento edition (Community or Commerce)
             // If it doesn't match, see if there is another patch listed for the other edition
             if (substr($file, 0, strlen($this->getEdition())) !== $this->getEdition()) {
-                $package = $packageKeys[1];
+                $package = next($packageKeys);
```

Comment:
There appear to be situations where `$packageKeys` looks like the following:
```
array(1) {
  [1]=>
  string(29) "magento/inventory-metapackage"
}
```
and `next()` returns `false`, which then results in the following error:
```
php bin/magento elgentos:quality-patches:convert
MDVA-34850


In ErrorHandler.php line 61:
                                                                                                                                                 
  Notice: Array to string conversion in /var/www/html/vendor/elgentos/magento2-composer-quality-patches/Console/Command/Convert.php on line 255  
                                                                                                                                                 

elgentos:quality-patches:convert
```

For the moment, I've replaced:
`$package = next($packageKeys);`

With:
`if (count($packageKeys) > 1) $package = next($packageKeys);`

And the `composer.quality-patches.json` file is generated, and the root `composer.json` file is updated.

Reply:
So only this?

```
            if (substr($file, 0, strlen($this->getEdition())) !== $this->getEdition()) {
-                $package = next($packageKeys);
+                if (count($packageKeys) > 1) $package = next($packageKeys);
                $description = array_key_first($this->patchesJsonContent[$patchId][$package]);
```

This would break if the `count($packageKeys)` isn't larger than 1, right?

Reply:
Hey @peterjaap , yes that's all that I changed. I also didn't take the scenario of `count($packageKeys)` not being larger than 1, and had presumed that `$package = reset($packageKeys)` [here](https://github.com/elgentos/magento2-composer-quality-patches/blob/fixes/Console/Command/Convert.php#L245) would reset the array pointer if there was at least one element.

The problem that I've ran into right now, is without the change that I've made, I'm already running into issues where `$package` has a value of `false` from the call to `reset()` , and when `$package` is used to access an array [here](https://github.com/elgentos/magento2-composer-quality-patches/blob/fixes/Console/Command/Convert.php#L254), the value of `false` results in the error I reported.

I think in both the case of what you've already written, and in the case of my suggested change, if `$package` is ever `false`, then there will be an `Undefined offset` error. It might make sense to bake in some defensiveness around some of the attempted array access.

If you're open to contributions, I am more than willing to create a branch off of your `fixes` branch, and submit a PR with some suggested changes?

Reply:
I'd be very happy to receive a PR! :)